tests/cli.rs

```diff
@@ -0,0 +1,295 @@
+#![cfg(any(target_os = "windows", target_os = "macos", target_os = "linux"))]
+use std::env;
+use std::fs::File;
+use std::io::{BufRead, BufReader};
+use std::path::{Path, PathBuf};
+use std::process::{Child, ChildStderr, Command, Stdio};
+
+use anyhow::{Context, Result};
+use rand::RngCore;
+use tempfile::tempdir;
+
+const ADDR: &str = "127.0.0.1:0";
+
+fn make_rand_file(size: usize, path: &Path) -> Result<()> {
+    let mut content = vec![0u8; size];
+    rand::thread_rng().fill_bytes(&mut content);
```

Comment:
please use `rand::rngs::StdRng::seed_from_u64(1)` to have the same randomness in every test run, otherwise if there is any issue it will be very hard to reproduce